futures-core/src/task/__internal/atomic_waker.rs

```diff
@@ -326,9 +340,7 @@ impl AtomicWaker {
                 // Release the lock
                 self.state.fetch_and(!(WAKING | HOLDS), Release);
 
-                if let Some(waker) = waker {
-                    waker.wake();
-                }
+                return waker;
```

Comment:
clippy is mad about your tail-position `return`s

Reply:
this is the Good Code but i'll change it